registry/service.go

```diff
@@ -71,7 +71,9 @@ func (s *Service) Auth(ctx context.Context, authConfig *registry.AuthConfig, use
 	// Lookup endpoints for authentication using "LookupPushEndpoints", which
 	// excludes mirrors to prevent sending credentials of the upstream registry
 	// to a mirror.
-	endpoints, err := s.LookupPushEndpoints(registryHostName)
+	s.mu.RLock()
+	endpoints, err := s.lookupV2Endpoints(registryHostName, false)
```

Comment:
The comment needs a matching update.

Reply:
Oh! Good catch; I initially kept it to use the "push endpoints", but then thought it may be clearer to remove that extra indirect and make it match the "except for mirrors" part.

Reply:
Updated 👍